src/Juvix/Core/Parameterisations/Naturals.hs

```diff
@@ -25,7 +26,14 @@ data NatVal
   | Sub -- f subtraction
   | Mul -- f multiplication
   | Curried NatVal Natural
-  deriving (Show, Eq)
+  deriving (Eq)
+
+instance Show NatVal where
+  show (Natural x) = " Nat " <> Text.Show.show x <> " "
```

Comment:
Not sure if the last space here is overkill...

Reply:
I believe it is so.

we get results like
```haskell
"(+ 3 4)" should parse as Elim (App (App (Prim  + ) (Elim (Prim  Nat 3 ))) (Elim (Prim  Nat 4 ))):
```

We should end up with
```haskell
"(+ 3 4)" should parse as Elim (App (App (Prim +) (Elim (Prim Nat 3 ))) (Elim (Prim Nat 4))):